ui/app/services/breadcrumbs.js

```diff
@@ -1,45 +1,20 @@
-import { getOwner } from '@ember/application';
-import Service, { inject as service } from '@ember/service';
-import { computed } from '@ember/object';
-import classic from 'ember-classic-decorator';
+import Service from '@ember/service';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import { schedule } from '@ember/runloop';
 
-@classic
-export default class BreadcrumbsService extends Service {
-  @service router;
+export default class BucketService extends Service {
+  @tracked crumbs = [];
 
-  // currentURL is only used to listen to all transitions.
-  // currentRouteName has all information necessary to compute breadcrumbs,
-  // but it doesn't change when a transition to the same route with a different
-  // model occurs.
-  @computed('router.{currentURL,currentRouteName}')
-  get breadcrumbs() {
-    const owner = getOwner(this);
-    const allRoutes = (this.get('router.currentRouteName') || '')
-      .split('.')
-      .without('')
-      .map((segment, index, allSegments) => allSegments.slice(0, index + 1).join('.'));
-
-    let crumbs = [];
-    allRoutes.forEach(routeName => {
-      const route = owner.lookup(`route:${routeName}`);
-
-      // Routes can reset the breadcrumb trail to start anew even
-      // if the route is deeply nested.
-      if (route.resetBreadcrumbs) {
-        crumbs = [];
-      }
-
-      // Breadcrumbs are either an array of static crumbs
-      // or a function that returns breadcrumbs given the current
-      // model for the route's controller.
-      let breadcrumbs = route.breadcrumbs || [];
-      if (typeof breadcrumbs === 'function') {
-        breadcrumbs = breadcrumbs(route.get('controller.model')) || [];
-      }
-
-      crumbs.push(...breadcrumbs);
+  @action registerBreadcrumb(crumb) {
+    schedule('actions', this, () => {
+      this.crumbs = [...this.crumbs, crumb];
     });
+  }
+
+  @action deregisterBreadcrumb(crumb) {
+    const newCrumbs = this.crumbs.filter(c => c !== crumb);
```

Comment:
We need to point to a new reference in memory since this is an array otherwise the tag won't be dirtied and a re-render won't trigger when we push to the array.

Reply:
Ah good call.